src/testing/builder/EventModelBuilder.js

```diff
@@ -0,0 +1,95 @@
+// @flow
+
+import { DateModel, EventModel, LocationModel } from '@integreat-app/integreat-api-client'
+import moment from 'moment-timezone'
+import seedrandom from 'seedrandom'
+import type { FileCacheStateType } from '../../modules/app/StateType'
+import hashUrl from '../../modules/endpoint/hashUrl'
+
+const MAX_PREDICTABLE_VALUE = 6
+
+class EventModelBuilder {
+  _eventCount: number
+  _seed: string
+
+  constructor (seed: string, eventCount: number) {
+    this._seed = seed
+    this._eventCount = eventCount
+  }
+
+  predictableNumber (index: number, max: number = MAX_PREDICTABLE_VALUE): number {
+    return seedrandom(index + this._seed)() * max
+  }
+
+  build (): Array<EventModel> {
+    return this.buildAll().map(all => all.event)
+  }
+
+  buildResources (): { [path: string]: FileCacheStateType } {
+    return this.buildAll().reduce((result, { path, resources }) => {
+      result[path] = resources
+      return result
+    }, {})
+  }
+
+  createResource (url: string, index: number, lastUpdate: moment): FileCacheStateType {
+    const hash = hashUrl(url)
+    return {
+      [url]: {
+        filePath: `path/to/documentDir/resource-cache/v1/some-city/files/${hash}.png`,
+        lastUpdate: moment(lastUpdate).add(this.predictableNumber(index), 'days'),
+        hash
+      }
+    }
+  }
+
+  /**
+   * Builds the requested amount of events. Two builds with an identical seed will yield equal events.
+   * Note that they are not identical. Furthermore instances of external classes like `moment` or `EventModel` maybe are
+   * not equal when comparing all the properties.
+   *
+   * @returns The events and the corresponding resource cache
+   */
+  buildAll (): Array<{ path: string, event: EventModel, resources: { [path: string]: FileCacheStateType } }> {
+    return Array.from({ length: this._eventCount }, (x, index) => {
+      const mockDate = moment('2015-01-01T00:00:00.000Z', moment.ISO_8601)
+      const startDate = moment(mockDate.add(this.predictableNumber(index), 'years').toISOString(), moment.ISO_8601)
+      const endDate = moment(mockDate.add(this.predictableNumber(index), 'hours').toISOString(), moment.ISO_8601)
+      const lastUpdate = moment(mockDate.subtract(this.predictableNumber(index), 'months').toISOString(), moment.ISO_8601)
+      const path = `/augsburg/en/events/event${index}`
+      return {
+        path,
+        event: new EventModel({
+          id: index + 1,
```

Comment:
why is the +1?

Reply:
Events do not have the id 0 afaik. The id 0 is special in WP.

Reply:
In fact, EventModel as well as CategoryModel don't have any numeric ids :D

Reply:
Remove numeric id and add `hash`